src/multio/action/scale/Scale.cc

```diff
@@ -0,0 +1,105 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+#include "multio/action/scale/Scale.h"
+
+#include "eckit/exception/Exceptions.h"
+#include "eckit/log/Log.h"
+
+
+#include "multio/LibMultio.h"
+#include "multio/config/ComponentConfiguration.h"
+#include "multio/message/Glossary.h"
+#include "multio/message/Message.h"
+#include "multio/util/PrecisionTag.h"
+
+#include "multio/action/scale/Mapping.h"
+#include "multio/action/scale/Scaling.h"
+
+
+namespace multio::action {
+
+using message::glossary;
+
+Scale::Scale(const ComponentConfiguration& compConf) :
+    ChainedAction(compConf), scaling_{compConf}, mapping_{compConf}, paramsToScale_{} {
+
+    const auto mappings = compConf.parsedConfig().has("mapping-definition")
+                            ? compConf.parsedConfig().getSubConfigurations("mapping-definition")
+                            : throw eckit::SeriousBug{"Scaling information not specified in plan", Here()};
+    ;
+
+    if (!mappings.empty()) {
+        for (const auto& mapping : mappings) {
+            auto matcher = mapping.getSubConfiguration("case");
+            paramsToScale_.insert(matcher.getString("param-is"));
+        }
+    }
+    std::cout << paramsToScale_ << std::endl;
+}
+
+void Scale::executeImpl(message::Message msg) {
+    if (msg.tag() != message::Message::Tag::Field) {
+        executeNext(std::move(msg));
+        return;
+    }
+
+    std::string cparam{"xxx"};
+    if (auto param = msg.metadata().getOpt<std::string>(glossary().param); param) {
```

Comment:
I don't think we need the `; param` bit. If we use assignment in the `if`, the result is whether `param` has a valid value or not. But this practice is, anyway, questionable. But I'm personally fine with it.

Reply:
This I got from Mirco as he was working on similar logic for his statistic class. I did some reading on this. It's true since I only use param and paramID within the if clause it should work but I think with cpp 17 the recommentation is to use as this?